datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -0,0 +1,133 @@
+package com.sequenceiq.datalake.service.sdx;
+
+import static com.sequenceiq.cloudbreak.common.exception.NotFoundException.notFound;
+import static com.sequenceiq.sdx.api.model.SdxClusterShape.ENTERPRISE;
+
+import java.util.Locale;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.WebApplicationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.StackV4Endpoint;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.StackScaleV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackV4Response;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.instancegroup.InstanceGroupV4Response;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareInternalCrnGeneratorFactory;
+import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.common.exception.WebApplicationExceptionMessageExtractor;
+import com.sequenceiq.datalake.entity.DatalakeInstanceGroupScalingDetails;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.flow.SdxReactorFlowManager;
+import com.sequenceiq.datalake.repository.SdxClusterRepository;
+import com.sequenceiq.datalake.service.EnvironmentClientService;
+import com.sequenceiq.environment.api.v1.environment.model.response.DetailedEnvironmentResponse;
+import com.sequenceiq.environment.api.v1.environment.model.response.EnvironmentStatus;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+import com.sequenceiq.sdx.api.model.DatalakeHorizontalScaleRequest;
+
+@Service
+public class SdxHorizontalScalingService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxHorizontalScalingService.class);
+
+    @Inject
+    private SdxClusterRepository sdxClusterRepository;
+
+    @Inject
+    private SdxReactorFlowManager sdxReactorFlowManager;
+
+    @Inject
+    private StackV4Endpoint stackV4Endpoint;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    @Inject
+    private EnvironmentClientService environmentClientService;
+
+    @Inject
+    private RegionAwareInternalCrnGeneratorFactory regionAwareInternalCrnGeneratorFactory;
+
+    @Inject
+    private WebApplicationExceptionMessageExtractor webApplicationExceptionMessageExtractor;
+
+    public FlowIdentifier horizontalScaleDatalake(String name, DatalakeHorizontalScaleRequest scaleRequest) {
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        if (!entitlementService.isDatalakeHorizontalScaleEnabled(accountId)) {
+            throw new BadRequestException("Datalake horizontal scale is not enabled");
```

Comment:
Might be better to add the accountId just for the sake of clarity.
Datalake horizontal scale is not enabled for account {}.